tests/kafkatest/services/console_consumer.py

```diff
@@ -122,47 +123,68 @@ def __init__(self, context, num_nodes, kafka, topic, security_protocol=None, new
         }
 
         self.consumer_timeout_ms = consumer_timeout_ms
+        for node in self.nodes:
+            node.version = version
 
         self.from_beginning = from_beginning
         self.message_validator = message_validator
         self.messages_consumed = {idx: [] for idx in range(1, num_nodes + 1)}
         self.client_id = client_id
+        self.security_protocol = security_protocol
+
+        # Validate a few configs
+        if self.new_consumer is None:
+            self.new_consumer = self.security_protocol == SecurityConfig.SSL
+        if self.security_protocol == SecurityConfig.SSL and not self.new_consumer:
+            raise Exception("SSL protocol is supported only with the new consumer")
 
+    def prop_file(self, node):
+        """Return a string which can be used to create a configuration file appropriate for the given node."""
         # Process client configuration
-        self.prop_file = self.render('console_consumer.properties', consumer_timeout_ms=self.consumer_timeout_ms, client_id=self.client_id)
+        prop_file = self.render('console_consumer.properties')
+        if hasattr(node, "version") and node.version <= LATEST_0_8_2:
+            # in 0.8.2.X and earlier, console consumer does not have --timeout-ms option
+            # instead, we have to pass it through the config file
+            prop_file += "consumer.timeout.ms=" + str(self.consumer_timeout_ms)
```

Comment:
Do we need a newline here, either at the beginning or the end? Depending on who happens to edit the `console_consumer.properties` file last and whether their editor leaves newlines at the end of files, it seems like this could break.